feat(📦): Add toggle runner

dsznajder opened the pull request:
[Toggle Preview](https://snack.expo.io/@dsznajder/toggle-animation-runner)

I've added toggle runner as I think this can be useful for people when you would like to run animations from `onPress` event

wcandillon commented:
The implementation looks identical to `useTransition` should we merge the two together?

dsznajder commented:
@wcandillon As I'm thinking of this one, I think this is useless :D But we are lack of documentation, and that's why I've build this one. I will focus more on some `Example` app instead ;)

wcandillon commented:
you could argue that while useTransition need the state to come from JS, your proposed implementation works for a change in an animation node which is in my opinion equally useful.
I'm just thinking about function names that would make sense for these two functions.

dsznajder commented:
@wcandillon I'm not sure about naming, I'm also concern about using 2 "state" values.

wcandillon commented:
And so `useTransition` would use this function in its implementation right? `useTransition` would just connect a JS State to this function.

dsznajder commented:
@wcandillon please take a look at https://snack.expo.io/@dsznajder/toggle-animation-runner

I changed useTransition to use `toggle`. I'm not sure if it has same effect but seems to be working fine ;)

dsznajder commented:
@wcandillon I've changed `useTransition` implementation to use `toggle` instead. Please review in free time ;)

wcandillon commented on README.md:
```diff
@@ -295,8 +295,43 @@ decay({ clock: Clock, value: Node, velocity: Node, deceleration: number} ): Node
+  block([
+    toggle({
+      value: translationY,
+      openValue: open,
```
Would it make sense to have only one value that is either one or zero?

dsznajder replied:
Yes, it would. However, I thought about readability. We can go for 1 value and after animation ends just set it State.END. What do you think?

dsznajder commented:
@wcandillon I've changed it to only one value. Also, added example to not confuse people how to properly use it. I hope it's enough ;)
I'm also working on `Example` app to showcase every function but it will take some time ;)

wcandillon commented:
@dsznajder I found a serious bug in `useTransition()` and added `useToggle()` based on that. Is there still a use case for `toggle()`? Please let me know.

dsznajder commented:
@wcandillon You still have to change some state if you would like to trigger it.

wcandillon commented:
Right! makes sense.

On Sun, Oct 20, 2019 at 8:41 AM Damian Sznajder
<notifications@github.com> wrote:
>
> @wcandillon You still have to change some state if you would like to trigger it.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.

wcandillon commented on src/AnimationRunners.ts:
```diff
@@ -248,3 +250,53 @@ export const loop = (loopConfig: LoopProps) => {
+          to
+        })
+      ),
+      cond(not(clockRunning(clock)), [set(toggleState, GestureState.END)])
```
@dsznajder is GestureState.END involved here?

dsznajder replied:
I just used it as `toggleState` when nothing should happen.

wcandillon replied:
Why should this function have a dependency with `GestureState`?

dsznajder replied:
@wcandillon I just use it as some descriptive value. I can use just some number (i.e. -1). What do you think?

wcandillon commented:
@dsznajder I love it! Thank you so much,

wcandillon commented:
:tada: This PR is included in version 8.4.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/react-native-redash)
- [GitHub release](https://github.com/wcandillon/react-native-redash/releases/tag/v8.4.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
